scripts/lib/sizefmt.py

```diff
@@ -0,0 +1,155 @@
+#!/bin/python
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+#
+# Converts human readable numbers to and from numbers.
+#
+##############################################################################
+
+import re
+import sys
+
+KB  = 1000
+MB  = 1000 * KB
+GB  = 1000 * MB
+TB  = 1000 * GB
+PB  = 1000 * TB
+
+KiB = 1024
+MiB = 1024 * KiB
+GiB = 1024 * MiB
+TiB = 1024 * GiB
+PiB = 1024 * TiB
+
+SIZES = {
+    'b': 1, 'kb': KiB, 'mb': MiB, 'gb': GiB, 'tb': TiB, 'pb': PiB,
+    'B': 1, 'Kb': KB, 'Mb': MB, 'Gb': GB, 'Tb': TB, 'Pb': PB,
+}
+decimapAbbrs = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+binaryAbbrs = ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB']
+sizeRegex = re.compile("^([-]?)(\d+(.\d+)?) ?([KkMmGgTtPp])?(i)?([Bb])?$")
+
+def parse_size(size):
+    """
+    Parses the human-readable size string into the amount it represents
+    as a float value. Upper case units are base 1000, lowercase base 1024.
+    If the unit has "i" in the middle it uses base 1024.
+
+    >>> parse_size("10K")
+    10000.0
+    >>> parse_size("-10K")
+    -10000.0
+    >>> parse_size("10 KB")
+    10000.0
+    >>> parse_size("10 G")
+    10000000000.0
+    >>> parse_size("10GB")
+    10000000000.0
+    >>> parse_size("10k")
+    10240.0
+    >>> parse_size("-10k")
+    -10240.0
+    >>> parse_size("10 kb")
+    10240.0
+    >>> parse_size("10gb")
+    10737418240.0
+    >>> parse_size("10g")
+    10737418240.0
+    >>> parse_size("10GiB")
+    10737418240.0
+    """
+    match = sizeRegex.match(size)
+    if not match:
+        return None
+    neg = -1 if match.group(1) else 1
```

Comment:
I wonder if it would help to have the regex groups enumerated in a comment.  It might make it easier on the next dev who works on this.  (e.g. `group 1 : +/-`, `group 2 : digits`, etc...)